src/rust/engine/process_execution/src/local.rs

```diff
@@ -74,15 +74,19 @@ impl CommandRunner {
     output_files_future
       .join(output_dirs_future)
       .and_then(|(output_files_stats, output_dirs_stats)| {
-        let paths: Vec<_> = output_files_stats
+        let mut paths: Vec<_> = output_files_stats
           .into_iter()
-          .chain(output_dirs_stats.into_iter().map(Some))
+          .filter_map(|v| v)
+          .chain(output_dirs_stats.into_iter())
           .collect();
+        // Sort both for consistent results and to allow deduping.
```

Comment:
I think that we would get this for free if we used `globs` to capture these, similar to how we capture directories. And we could probably do it with one invoke.

I can't remember why we did this as "use globs for dirs" and "use literals for literals", but doing it all with globs might be easier.

Reply:
Done :)